src/lib/djerba/plugins/genomic_landscape/render.py

```diff
@@ -11,15 +11,25 @@ def assemble_biomarker_plot(self,biomarker,plot):
         cell = template.format(biomarker,plot)
         return(cell)
 
-    def biomarker_table_rows(self, biomarkers, can_report_hrd, can_report_msi):
+    def biomarker_table_rows(self, biomarkers, can_report_hrd, cant_report_hrd_reason, can_report_msi):
         rows = []
         for marker, info in biomarkers.items():
-            if marker == "HRD" and not can_report_hrd:
-                cells = [
-                    hb.td(info[constants.ALT]),
-                    hb.td("NA"),
-                    hb.td("Cancer cell content below threshold to evaluate HRD; must be &#8805;50&#37; for FFPE samples, &#8805;30&#37; otherwise")
-                ]
+            if marker == "HRD" and not can_report_hrd and cant_report_hrd_reason:
+                if cant_report_hrd_reason == "purity":
```

Comment:
I'm OK with the logic of reporting purity below threshold if that applies, then coverage below threshold if it applies, then reporting all is well. The output will not be incorrect, and I don't think we need extra code to handle the case of "purity too low" and "coverage too high" at the same time.